replica/table.cc

```diff
@@ -1478,11 +1478,16 @@ future<std::unordered_map<sstring, table::snapshot_details>> table::get_snapshot
                 continue;
             }
 
-            lister::scan_dir(snapshots_dir,  { directory_entry_type::directory }, [this, datadir, &all_snapshots] (fs::path snapshots_dir, directory_entry de) {
+            lister::scan_dir(snapshots_dir,  { directory_entry_type::directory }, [this, datadir, &all_snapshots] (fs::path snapshots_dir, directory_entry de) -> future<> {
                 auto snapshot_name = de.name;
-                all_snapshots.emplace(snapshot_name, snapshot_details());
-                return lister::scan_dir(snapshots_dir / fs::path(snapshot_name),  { directory_entry_type::regular }, [this, datadir, &all_snapshots, snapshot_name] (fs::path snapshot_dir, directory_entry de) {
-                    return io_check(file_size, (snapshot_dir / de.name).native()).then([this, datadir, &all_snapshots, snapshot_name, snapshot_dir, name = de.name] (auto size) {
+                auto snapshot_dir = snapshots_dir / snapshot_name;
+                auto sd = co_await io_check(file_stat, snapshot_dir.native(), follow_symlink::no);
```

Comment:
Do we even expect any symbolic links in this directory? AFAIK the only links we expect are hard links to sstable files (for which this flag should have no effect). I'm not even sure how we should count symbolic links if they were part of a snapshot...

Reply:
No, we don't expect symbolic links anywhere in our file hierarchy.
But we do have to provide a value for this arg since there is no default for it (on purpose).
If I'd pass `follow_symlink::yes` I think you would have asked the same question :)